--- gap4r4/pkg/xgap/PackageInfo.g.orig	2009-06-12 19:22:11.000000000 -0300
+++ gap4r4/pkg/xgap/PackageInfo.g	2009-06-12 19:23:22.000000000 -0300
@@ -242,7 +242,7 @@
 ## tests of other packages, as given above, will be done automatically and
 ## need not be included here.)
 # AvailabilityTest := ReturnTrue,
-AvailabilityTest := function() return GAPInfo.CommandLineOptions.p; end,
+AvailabilityTest := ReturnFalse,
 
 ##  If the default banner does not suffice then provide a string that is
 ##  printed when the package is loaded (not when it is autoloaded or if
@@ -253,7 +253,7 @@
 ##  started.  This should usually be 'false'. Say 'true' only if your package 
 ##  provides some improvements of the GAP library which are likely to enhance 
 ##  the overall system performance for many users.
-Autoload := true,
+Autoload := false,
 
 ##  *Optional*, but recommended: path relative to package root to a file which 
 ##  contains as many tests of the package functionality as sensible.
